src/MOI_wrapper.jl

```diff
@@ -0,0 +1,2708 @@
+# Follows the implementation of the Gurobi MOI Wrapper
+# https://github.com/JuliaOpt/Gurobi.jl/blob/master/src/MOI_wrapper.jl
+
+import MathOptInterface
+
+const MOI = MathOptInterface
+const CleverDicts = MOI.Utilities.CleverDicts
+
+@enum(VariableType, CONTINUOUS, BINARY, INTEGER, SEMIINTEGER, SEMICONTINUOUS)
+@enum(BoundType, NONE, LESS_THAN, GREATER_THAN, LESS_AND_GREATER_THAN, INTERVAL, EQUAL_TO)
+@enum(ObjectiveType, SINGLE_VARIABLE, SCALAR_AFFINE, SCALAR_QUADRATIC)
+
+const SCALAR_SETS = Union{
+    MOI.GreaterThan{Float64}, MOI.LessThan{Float64},
+    MOI.EqualTo{Float64}, MOI.Interval{Float64}
+}
+
+mutable struct VariableInfo
+    index::MOI.VariableIndex
+    column::Int
+    bound::BoundType
+    type::VariableType
+    start::Union{Float64, Nothing}
+    name::String
+    # Storage for constraint names associated with variables because Gurobi

```

Comment:
Should we replace Gurobi by Xpress here?

Reply:
Pending to check if it applied to Xpress. @joaquimg do we need this too?

Reply:
Yes we need. Variable bound are not proper constraints in xpress also.